File: problems/daily/Java/706. Design HashMap.java
class MyHashMap {
    private static final int s = 1000;
    private List<int[]>[] map;
    public MyHashMap() {
        map = new ArrayList[s];
        for(int i = 0;i < s; i++){
            map[i]= new ArrayList<>();
        }
    }
    
    public void put(int key, int value) {
        int id = key % s;
        List<int[]> tent = map[id];
        for(int[] i:tent){
            if(i[0] == key){
                i[1] = value;
                return;
            }
        }
        tent.add(new int[]{key,value});
    }
    
    public int get(int key) {
        int id = key % s;
        List<int[]> tent = map[id];
        for(int[] i:tent){
            if(i[0] == key){
                return i[1];
            }
        }
        return -1;
    }
    
    public void remove(int key) {
        int id = key % s;
        List<int[]>tent = map[id];
        for(int k = 0; k < tent.size();k++){
            int[] i = tent.get(k);
            if(i[0] == key){
                tent.remove(k);
                return;
            }
        }
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */

File: problems/daily/Python3/706. Design HashMap.py
class MyHashMap:

    def __init__(self):
        self.ans = [None] * 1000001
    def put(self, key: int, value: int) -> None:
        self.ans[key] = value
    def get(self, key: int) -> int:
        value = self.ans[key]
        return value if value != None else -1
    def remove(self, key: int) -> None:
        self.ans[key] = None

# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)